py-janus: new package

teaguesterling opened the pull request:
Adding latest version of `py-janus` package as well as an older version needed for a new package.

teaguesterling committed:
py-uvloop: add v3.8.14, v3.9.15, v3.10.3 and dependencies

teaguesterling committed:
rollback

teaguesterling committed:
py-janus: add v.0.7.0,v1.0.0

tldahlgren commented on var/spack/repos/builtin/packages/py-janus/package.py:
```diff
@@ -0,0 +1,23 @@
+    version("0.7.0", sha256="f10dcf5776e8d49cc30ec86d5eb7268eeec39abaa24fe0332ee8fb8fa3611845")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
The `pyproject.toml` files for both versions indicate the dependency needs to be constrained to `py-setuptools@51:`.

tldahlgren commented on var/spack/repos/builtin/packages/py-janus/package.py:
```diff
@@ -0,0 +1,23 @@
+
+    version("1.0.0", sha256="df976f2cdcfb034b147a2d51edfc34ff6bfb12d4e2643d3ad0e10de058cb1612")
+    version("0.7.0", sha256="f10dcf5776e8d49cc30ec86d5eb7268eeec39abaa24fe0332ee8fb8fa3611845")
+
```
Both versions also appear to require `py-wheel@0.36:`.

tldahlgren reviewed (changes requested):
Confirmed the homepage and version sha256s but the `pyproject.toml` files indicate the dependencies need to be further constrained.

teaguesterling committed:
Update package.py

Incorporating changes from review: https://github.com/spack/spack/pull/44520#pullrequestreview-2095028464

teaguesterling committed:
Update package.py

tldahlgren reviewed (approved):
Thanks.
